Makefile

```diff
@@ -548,103 +569,125 @@ $(TOOLCHAIN):
 	git config --global core.autocrlf false
 	mkdir -p $(TOOLCHAIN)
 
+$(PATCH_DIR):
+ifeq ($(GCC_VERSION),4.9.2)
+	cd $(XTDLP)/$(GCC_DIR)
+	patch -p1 < $(PATCH_DIR)/0001-WIP-don-t-bring-extra-u-int_least32_t-into-std.patch
+endif
+
 # GMP
 $(XTDLP)/$(GMP_DIR): $(XTDLP)/$(GMP_TAR)
 	mkdir -p $(XTDLP)/$(GMP_DIR)
 	$(UNTAR) $(XTDLP)/$(GMP_TAR) -C $(XTDLP)/$(GMP_DIR)
-	mv $(XTDLP)/$(GMP_DIR)/gmp-*/* $(XTDLP)/$(GMP_DIR)
+	rm -rf $(XTDLP)/$(GMP_DIR)/build
 
 $(XTDLP)/$(GMP_DIR)/build: $(XTDLP)/$(GMP_DIR)
 	mkdir -p $(XTDLP)/$(GMP_DIR)/build/
-	cd $(XTDLP)/$(GMP_DIR)/build/; ../configure --prefix=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
-	make -C $(XTDLP)/$(GMP_DIR)/build/	
+	cd $(XTDLP)/$(GMP_DIR)/build/; ../$(CONFIGURE) --prefix=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
+	$(MAKE) -C $(XTDLP)/$(GMP_DIR)/build/
+	@touch $(XTDLP)/$(GMP_DIR)/build
 
 $(XTBP)/gmp: $(XTDLP)/$(GMP_DIR)/build
-	make install -C $(XTDLP)/$(GMP_DIR)/build/
+	$(MAKE) $(INSTALL) -C $(XTDLP)/$(GMP_DIR)/build/
+	@touch $(XTBP)/gmp
 
-# MPFR	
+# MPFR
 $(XTDLP)/$(MPFR_DIR): $(XTDLP)/$(MPFR_TAR)
 	mkdir -p $(XTDLP)/$(MPFR_DIR)
 	$(UNTAR) $(XTDLP)/$(MPFR_TAR) -C $(XTDLP)/$(MPFR_DIR)
-	mv $(XTDLP)/$(MPFR_DIR)/mpfr-*/* $(XTDLP)/$(MPFR_DIR)
+	rm -rf $(XTDLP)/$(MPFR_DIR)/build
 
 $(XTDLP)/$(MPFR_DIR)/build: $(XTDLP)/$(MPFR_DIR)
 	mkdir -p $(XTDLP)/$(MPFR_DIR)/build
-	cd $(XTDLP)/$(MPFR_DIR)/build/; ../configure --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
-	make -C $(XTDLP)/$(MPFR_DIR)/build/	
+	cd $(XTDLP)/$(MPFR_DIR)/build/; ../$(CONFIGURE) --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
+	$(MAKE) -C $(XTDLP)/$(MPFR_DIR)/build/
+	@touch $(XTDLP)/$(MPFR_DIR)/build
```

Comment:
If you do the make the first time, then you have a new build directory, 
so you have a clear dependency source older then build.
But the build directory is part the source, so after the make you do not 
know what is older, what newer.

Specially the configure process changes sources.
Another problem is, that I don't know if the timestamp of a parent 
directory (e.g. build) will always reflect the timestamp of subdirs or 
even files in subdirs.

At least, if there was any fault and you have to try the make once more 
(without make clean) then the dependency has changed or not.

After all my changes (touch) I got a make process, which will not make 
the targets again and again, like it was before.

Am 23.12.2015 um 01:39 schrieb Dmitry Kireev:
> In Makefile
> <https://github.com/kireevco/esp-alt-sdk/pull/8#discussion_r48314116>:
>
>>
>>  $(XTDLP)/$(MPFR_DIR)/build: $(XTDLP)/$(MPFR_DIR)
>>  	mkdir -p $(XTDLP)/$(MPFR_DIR)/build
>> -	cd $(XTDLP)/$(MPFR_DIR)/build/; ../configure --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
>> -	make -C $(XTDLP)/$(MPFR_DIR)/build/	
>> +	cd $(XTDLP)/$(MPFR_DIR)/build/; ../$(CONFIGURE) --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
>> +	$(MAKE) -C $(XTDLP)/$(MPFR_DIR)/build/
>> +	@touch $(XTDLP)/$(MPFR_DIR)/build
>
> Can you please explain why this is needed?
> wouldn't |mkdir -p $(XTDLP)/$(MPFR_DIR)/build| be enough?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kireevco/esp-alt-sdk/pull/8/files#r48314116>.
>